pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go

```diff
@@ -82,9 +82,16 @@ func mergeTransportSocketCluster(c *cluster.Cluster, cp *model.EnvoyFilterConfig
 				break
 			}
 		}
+		if tsmPatch == nil && len(c.GetTransportSocketMatches()) > 0 {
+			c.GetTransportSocketMatches()[0].TransportSocket = cpValueCast.TransportSocket
```

Comment:
not sure this is right.. if they have a transport socket match and the name doesn't match what should we actually do?

Reply:
Yeah. I think this is not correct. This might incorrectly replace another context. May be we should log a warning and skip?

Reply:
My concern was that this means you can effectively not use custom transport socket in istio auto mtls. Maybe that is ok. The user that complained about the issue was only using for listener which is un-ambiguous. I will remove that for now, it didn't work prior to the PR that broke things anyways, so its a new feature essentially